### DISCUSSION TITLE
DataGrid programmatically Focus

### DISCUSSION THREAD

**Discussion by DomenicoFormoso** [Megabit/Blazorise#5094] [Q&A]
Is it possible to give focus to the datagrid component programmatically?
for example:
```
  <DataGrid @ref="@myGrid"
 .... 
<TextEdit  KeyUp="@OnRicercaDescrizione" Autofocus></TextEdit>
.....

private async Task OnRicercaDescrizione(KeyboardEventArgs e)
{
    if (e.Key == "ArrowDown")
    {
        await myGrid. Focus
     }
}

```
Thanks

**Comment by David-Moreira** [comment]
No, it isn't. DataGrid isn't focusable by default. As most elements in web pages that aren't related to an operation / input.

So **Focus** is an operation that's usually available on inputs (like text or numeric inputs) or elements in a page that can trigger an operation, like buttons.

What is your use case for this?

**Reply by DomenicoFormoso** [reply]
Thank you for your answer. I have to create a reusable search mod with a TextEdit and a DataGrid, the focus is on the TextEdit which sets the selection of the datagrid by key.  After typing the key and pressing Return you get the list of records in the DataGrid. Now by pressing the Down key the focus switches to the datagrid which I can scroll with the up and down keys. 
For the moment I solved it like this, it seems to work well

```
<TextEdit @bind-Text="@cosaCerco" Placeholder="Articolo da cercare..." KeyUp="@OnRicercaDescrizione" Autofocus></TextEdit>

private async Task OnRicercaDescrizione(KeyboardEventArgs e)
{
    if (e.Key == "ArrowDown")
    {
        myGrid.Select(myGrid.Data.FirstOrDefault());
    }
}
```


Now I found a flaw in scrolling up and down when I set the Navigable="true" parameter

**Comment by DomenicoFormoso** [comment]
```
 <Column ColumnSize="ColumnSize.Is12.OnDesktop">
     <DataGrid @ref="@myGrid"
               TItem="Articolo"
               Data="@_lista"
               Narrow
               Navigable="true"
               @bind-SelectedRow="@selectedArticolo"
               Virtualize>
         <DataGridCommandColumn />
         <DataGridColumn Field="@nameof(Articolo.Idarticolointerno)" Caption="Interno" Sortable="false" />
         <DataGridColumn Field="@nameof(Articolo.Idarticolobarre)" Caption="Produttore" />
         <DataGridColumn Field="@nameof(Articolo.Descrizione)" Caption="Descrizione" />
         <DataGridColumn Field="@nameof(Articolo.Quantita)" TextAlignment="TextAlignment.End" Caption="Q.ta" />
         <DataGridColumn Field="@nameof(Articolo.Prezzodettaglio)" Caption="Prezzo" DisplayFormat="{0:C4}" TextAlignment="TextAlignment.End" DisplayFormatProvider="@System.Globalization.CultureInfo.GetCultureInfo("it-IT")">
         </DataGridColumn>
     </DataGrid>
 </Column>
```
I created a video showing the problem but how can I share it?
